dali/python/backend_impl.cc

```diff
@@ -273,35 +273,32 @@ void ExposeTensorList(py::module &m) { // NOLINT
       ----------
       )code")
     .def("as_array", [](TensorList<CPUBackend> &t) -> py::array {
+          void* raw_mutable_data;
+          std::string format;
+          size_t type_size;
+
           if (t.size() == 0) {
-            std::string format;
-            size_t type_size;
-            if (IsValidType(t.type())) {
-              format = FormatStrFromType(t.type());
-              type_size = t.type().size();
-            } else {
-              // Default is float
-              format = py::format_descriptor<float>::format();
-              type_size = sizeof(float);
-            }
-            // Based on default numpy empty array shape and strides
-            std::vector<ssize_t> shape(1, 0);
-            std::vector<ssize_t> strides(1, type_size);
-            return py::array(py::buffer_info(nullptr,
-                                             type_size,
-                                             std::move(format),
-                                             shape.size(),
-                                             shape,
-                                             strides));
+            raw_mutable_data = nullptr;
+          } else {
+            DALI_ENFORCE(IsValidType(t.type()), "Cannot produce "
+                "buffer info for tensor w/ invalid type.");
+            DALI_ENFORCE(t.IsDenseTensor(),
+                        "Tensors in the list must have the same shape");
+            raw_mutable_data = t.raw_mutable_data();
           }
 
-          DALI_ENFORCE(IsValidType(t.type()), "Cannot produce "
-              "buffer info for tensor w/ invalid type.");
-          DALI_ENFORCE(t.IsDenseTensor(),
-                      "Tensors in the list must have the same shape");
+          if (IsValidType(t.type())) {
+            format = FormatStrFromType(t.type());
+            type_size = t.type().size();
+          } else {
+            // Default is float
```

Comment:
Why? Is this something numpy does as well?

Reply:
Yes